@here/harp-mapview/lib/geometry/TileGeometryLoader.ts

```diff
@@ -246,16 +256,33 @@ export class SimpleTileGeometryLoader implements TileGeometryLoader {
         enabledKinds: GeometryKindSet | undefined,
         disabledKinds: GeometryKindSet | undefined
     ) {
+        if (this.isFinished) {
+            return;
+        }
         // If the tile is not ready for display, or if it has become invisible while being loaded,
         // for example by moving the camera, the tile is not finished and its geometry is not
         // created. This is an optimization for fast camera movements and zooms.
         const tile = this.tile;
         const decodedTile = this.m_decodedTile;
         this.m_decodedTile = undefined;
```

Comment:
Anybody knows why this was set (garbage collector, rejections)? This is more or less the reason of entire problems, the isLoading method fixes it using different member, but ideally I would remove this line and change isLoading getter to:
```
get isLoading(): boolean {
return this.m_decodedTile !== undefined && this.m_timeout !== undefined;
}
```

Reply:
I also wondered about this, and even created a ticket for it: HARP-8742

Even though it is removed, it still exists in the TileLoader, so we aren't saving any memory.

Reply:
I confirm that originally this was to save memory, so decoded tile is wiped after closure passed to SetTimeout disapears ... then `*GeometryLoaders` were born and everything changed.

Reply:
So the newest implementation will make it undefined only after TileGeometryLoaded has finished his job. Because decoded tile it stored elsewhere, currently it does not give us any optimization.